main.py

```diff
@@ -203,23 +336,27 @@ def process_app_data(metadata, cmd_string, input_filepath, stdout, stderr, retur
   metadata['application']['return_code'] = return_code
 
   cwd = os.getcwd()
+  work_cwd = os.path.join(cwd, DEFAULT_WORK_DIR)
+
+  # This is to write all files to work directory.  This should
+  # probably be created in the main function however.  
+  if not os.path.exists(work_cwd): 
+    os.mkdir(work_cwd)
 
   # For the stdin, stdout and stderr, write each to a file, hash it, and store 
   # the hash and filepath to the metadata
   if input_filepath:
-    saved_input_path = os.path.join(cwd,"in")
-    shutil.copyfile(input_filepath, saved_input_path)
```

Comment:
I'd sort of sent a note out about this yesterday when I was looking at the code.  I wanted to know why it was added because it looks like the <input file> is copied over to the "in" file.  This was also logging into the metadata files as "in" instead of the name of the input file.

I am guessing that it was to make sure that the file wasn't changed while we were in the process of using it?  I'm ok if you want to change the code and add it back and name it similar to the input file though!  Also is that the best practice or can we assume it is safe to just use the input file instead of creating an additional copy?